twill-yarn/src/test/java/org/apache/twill/filesystem/LocalLocationTest.java

```diff
@@ -40,4 +49,25 @@ protected UserGroupInformation createTestUGI() throws IOException {
     // In local location, UGI is not supported, hence using the current user as the testing ugi.
     return UserGroupInformation.getCurrentUser();
   }
+
+  @Test
+  public void testLastModified() throws IOException {
+    LocationFactory lf = new LocalLocationFactory(TEMP_FOLDER.newFolder());
+    Location location = lf.create("test1");
+    String message = "message";
+    try (OutputStream os = location.getOutputStream()) {
+      os.write(message.getBytes(StandardCharsets.UTF_8));
+    }
+    long initialModificationTimestamp = location.lastModified();
+
+    // Modify file, modified time should get updated.
```

Comment:
Add `Thread.sleep(1)` in case when FS is very fast

Reply:
Added.